plugins/services/src/js/utils/FrameworkUtil.js

```diff
@@ -47,17 +47,20 @@ const FrameworkUtil = {
    * @return {Object}               containing urls from the extracted query string
    * if all is not available in given object.
    */
-  extractImageUrls(packageImages) {
-    const images = Object.assign({}, packageImages);
+  extractImageUrls(packageImages = {}) {
+    const variants = ["icon-small", "icon-medium", "icon-large"];
 
-    Object.keys(images).forEach(image => {
-      const queryString = images[image].substring(
-        images[image].lastIndexOf("?")
-      );
-      images[image] = RouterUtil.getQueryStringInUrl(queryString).url;
-    });
+    return variants
+      .filter(variant => packageImages[variant] != null)
+      .reduce((acc, variant) => {
+        const queryString = packageImages[variant].substring(
+          packageImages[variant].lastIndexOf("?")
```

Comment:
Is it possible these urls won't have a querystring with a "url" key? How defensive should we be here?

Reply:
Good point, we actually don't extract them on the catalog page, just call through the proxy.